tools/cold-backup/run.sh

```diff
@@ -0,0 +1,252 @@
+        oc delete pod $__MPOD
+    else
+        __error "Missing required parameters"
+    fi
+}
+
+__compress() {
+    if [[ -n $BACKUP_NAME && -n $BACKUP_HOME ]]; then
+        echo "Compressing"
+        local current_dir=$(pwd)
+        cd $__TMP
+        zip -qr $BACKUP_NAME.zip *
+        mv $BACKUP_NAME.zip $BACKUP_HOME
+        cd $current_dir
+    else
+        __error "Missing required parameters"
+    fi
+}
+
+__uncompress() {
+    if [[ -n $BACKUP_NAME && -n $BACKUP_HOME ]]; then
+        echo "Uncompressing"
+        rm -rf $__TMP
+        unzip -qo $BACKUP_HOME/$BACKUP_NAME.zip -d $__TMP
+    else
+        __error "Missing required parameters"
+    fi
+}
+
+backup() {
+    __confirm $SOURCE_NS
+    mkdir -p $__TMP/resources $__TMP/data
+    oc project $SOURCE_NS
+
+    # export operator version
+    local op_pod="$(oc get pods | grep strimzi-cluster-operator | grep Running | cut -d " " -f1)"
+    oc exec -it $op_pod -- env | grep "STRIMZI_VERSION" > $__TMP/README
+
+    # export resources
+    __export "kafkas"
+    __export "kafkatopics"
+    __export "kafkausers"
+    __export "kafkabridges"
+    __export "kafkaconnectors"
+    __export "kafkaconnects"
+    __export "kafkaconnects2is"
+    __export "kafkamirrormaker2s"
+    __export "kafkamirrormakers"
+    __export "kafkarebalances"
+    # internal certificates and user secrets
+    __export "secrets" "strimzi.io/name=strimzi"
+    # custom configmap and secrets
+    for name in "${CUSTOM_CMS[@]}"; do
+        __export "cm/$name"
+    done
+    for name in "${CUSTOM_SECRETS[@]}"; do
+        __export "secret/$name"
+    done
+
+    # stop operator and statefulsets
+    oc scale deployment strimzi-cluster-operator --replicas 0
+    oc scale statefulset $CLUSTER_NAME-kafka --replicas 0
+    oc scale statefulset $CLUSTER_NAME-zookeeper --replicas 0
+    sleep 120
+
+    # for each PVC, rsync data from PV to backup
+    for (( i = 0; i < $ZOO_REPLICAS; i++ )); do
+        local pvc="data-$CLUSTER_NAME-zookeeper-$i"
+        local local_path="$__TMP/data/$pvc"
+        mkdir -p $local_path
+        __rsync $pvc $__MPOD:/data/. $local_path
+    done
+    for (( i = 0; i < $KAFKA_REPLICAS; i++ )); do
+        if [ $KAFKA_JBOD -gt 0 ]; then
+            for (( j = 0; j < $KAFKA_JBOD; j++ )); do
+                local pvc="data-$j-$CLUSTER_NAME-kafka-$i"
+                local local_path="$__TMP/data/$pvc"
+                mkdir -p $local_path
+                __rsync $pvc $__MPOD:/data/. $__TMP/data/$pvc
+            done
+        else
+            local pvc="data-$CLUSTER_NAME-kafka-$i"
+            local local_path="$__TMP/data/$pvc"
+            mkdir -p $local_path
+            __rsync $pvc $__MPOD:/data/. $__TMP/data/$pvc
+        fi
+    done
+
+    # start statefulsets and operator
+    oc scale statefulset $CLUSTER_NAME-zookeeper --replicas $ZOO_REPLICAS
+    oc scale statefulset $CLUSTER_NAME-kafka --replicas $KAFKA_REPLICAS
```

Comment:
You should not do this ... you should just scale the operator (or unpause the resource - see the other comment) and let the operator do this.